### PR TITLE
[RFR] Add more details and search, sort ...

### PR DESCRIPTION
**Pull request by BaptistePDJ**

### CODE REVIEW

File: project/src/Controller/ClientController.php [hunk 1]
```diff
@@ -21,6 +21,53 @@ public function index(ClientRepository $clientRepository): Response
...
+    {
+        return $this->render('client/search.html.twig', [
+            'clients' => $clientRepository->findBy(['name' => $request->get('searchName')]),
+        ]);
+    }
+
+    #[Route('/ordered', name: 'client_ordered', methods: ['GET'])]
+    public function ordered(Request $request, ClientRepository $clientRepository): Response
+    {
+
```
Comment (julienmilazzo) [inline project/src/Controller/ClientController.php]: Remove this break line

File: project/src/Controller/ClientController.php [hunk 2]
```diff
@@ -21,6 +21,53 @@ public function index(ClientRepository $clientRepository): Response
...
+                ]);
+            case 'enable' :
+                return $this->render('client/index.html.twig', [
+                    'clients' => $clientRepository->findBy([], ['enable' => 'ASC']),
+                ]);
+        }
+        return $this->render('client/search.html.twig', [
+            'clients' => $clientRepository->findAll(),
+        ]);
+    }
```
Comment (julienmilazzo) [inline project/src/Controller/ClientController.php]: You can use the new php 8 function : `match`
See : https://www.php.net/manual/en/control-structures.match.php#example-133

File: project/src/Controller/DomainNameController.php [hunk 3]
```diff
@@ -21,6 +21,65 @@ public function index(DomainNameRepository $domainNameRepository): Response
...
+                    'domain_names' => $domainNameRepository->findBy([], ['enable' => 'ASC']),
+                ]);
+            case 'site' :
+                return $this->render('domain_name/index.html.twig', [
+                    'domain_names' => $domainNameRepository->findBy([], ['site' => 'ASC']),
+                ]);
+        }
+        return $this->render('domain_name/search.html.twig', [
+            'domain_names' => $domainNameRepository->findAll(),
+        ]);
```
Comment (julienmilazzo) [inline project/src/Controller/DomainNameController.php]: Same here (https://www.php.net/manual/en/control-structures.match.php#example-133)

File: project/src/Controller/DomainNameController.php [hunk 4]
```diff
@@ -21,6 +21,65 @@ public function index(DomainNameRepository $domainNameRepository): Response
...
+    {
+        return $this->render('domain_name/search.html.twig', [
+            'domain_names' => $domainNameRepository->findBy(['url' => $request->get('searchUrl')]),
+        ]);
+    }
+
+    #[Route('/ordered', name: 'domain_name_ordered', methods: ['GET'])]
+    public function ordered(Request $request, DomainNameRepository $domainNameRepository): Response
+    {
+
```
Comment (julienmilazzo) [inline project/src/Controller/DomainNameController.php]: Remove this break line pls

File: project/src/Controller/ServerController.php [hunk 5]
```diff
@@ -21,6 +21,61 @@ public function index(ServerRepository $serverRepository): Response
...
+    {
+        return $this->render('server/search.html.twig', [
+            'servers' => $serverRepository->findBy(['name' => $request->get('searchName')]),
+        ]);
+    }
+
+    #[Route('/ordered', name: 'server_ordered', methods: ['GET'])]
+    public function ordered(Request $request, ServerRepository $serverRepository): Response
+    {
+
```
Comment (julienmilazzo) [inline project/src/Controller/ServerController.php]: Remove this pls

File: project/src/Controller/ServerController.php [hunk 6]
```diff
@@ -21,6 +21,61 @@ public function index(ServerRepository $serverRepository): Response
...
+                    'servers' => $serverRepository->findBy([], ['offer' => 'ASC']),
+                ]);
+            case 'enable' :
+                return $this->render('server/index.html.twig', [
+                    'servers' => $serverRepository->findBy([], ['enable' => 'ASC']),
+                ]);
+        }
+        return $this->render('server/search.html.twig', [
+            'servers' => $serverRepository->findAll(),
+        ]);
```
Comment (julienmilazzo) [inline project/src/Controller/ServerController.php]: Same here (https://www.php.net/manual/en/control-structures.match.php#example-133)

File: project/src/Controller/SiteController.php [hunk 7]
```diff
@@ -21,6 +21,40 @@ public function index(SiteRepository $siteRepository): Response
...
+                    'sites' => $siteRepository->findBy([], ['online' => 'ASC']),
+                ]);
+            case 'enable' :
+                return $this->render('site/index.html.twig', [
+                    'sites' => $siteRepository->findBy([], ['enable' => 'ASC']),
+                ]);
+        }
+        return $this->render('site/search.html.twig', [
+            'sites' => $siteRepository->findAll(),
+        ]);
```
Comment (julienmilazzo) [inline project/src/Controller/SiteController.php]: Same here (https://www.php.net/manual/en/control-structures.match.php#example-133)

File: project/templates/base.html.twig [hunk 8]
```diff
@@ -2,7 +2,7 @@
 <html>
     <head>
         <meta charset="UTF-8">
-        <title>{% block title %}Welcome!{% endblock %}</title>
+        <title>{% block title %}Bienvenue!{% endblock %}</title>
         <link rel="shortcut icon" href="{{ app.request.getSchemeAndHttpHost() }}/images/teds.png" />
         {# Run `composer require symfony/webpack-encore-bundle` to start using Symfony UX #}
```
Comment (julienmilazzo) [inline project/templates/base.html.twig]: Remove this comment

File: project/templates/domain_name/search.html.twig [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+                        <td>{{ domain_name.invoicedPrice }}</td>
+                        <td>{{ domain_name.renewalType }}</td>
+                        <td>{{ domain_name.renewalDate ? domain_name.renewalDate|date('Y-m-d') : '' }}</td>
+                        <td>{{ domain_name.site.name }}</td>
+                        <td>{{ domain_name.enable ? 'Oui' : 'Non' }}</td>
+                        <td>
+                            <a href="{{ path('domain_name_show', {'id': domain_name.id}) }}" class="btn btn-secondary" style="margin-bottom: 5px">Détails</a>
+                            <a href="{{ path('domain_name_edit', {'id': domain_name.id}) }}" class="btn btn-success">Éditer</a>
+                        </td>
+
```
Comment (julienmilazzo) [inline project/templates/domain_name/search.html.twig]: remove this break line

File: project/templates/footer.html.twig [hunk 10]
```diff
@@ -1,5 +1,5 @@
-<footer>
-    <div class="container pt-lg-5 pt-4">
+<footer style="background-color: whitesmoke">
```
Comment (julienmilazzo) [inline project/templates/footer.html.twig]: You can do it in the css file

File: project/src/Twig/AppExtension.php [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+    /**
+     * @param UrlGeneratorInterface $generator
+     * @param RequestStack $requestStack
+     */
+    public function __construct(UrlGeneratorInterface $generator, RequestStack $requestStack)
+    {
+        $this->generator = $generator;
+        $this->requestStack = $requestStack;
+    }
```
Comment (BaptistePDJ) [inline project/src/Twig/AppExtension.php]: you can remove this maybe

File: project/src/Twig/AppExtension.php [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+     * {@inheritDoc}
+     */
+    public function getFunctions()
+    {
+        return [
+            new TwigFunction('sort', [$this, 'howIsSorted']),
+        ];
+    }
+
+    public function howIsSorted($orderedType, $expectedType, $orderBy = null)
```
Comment (julienmilazzo) [inline project/src/Twig/AppExtension.php:27]: Phpdoc please ;)

File: project/src/Twig/AppExtension.php [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        $returnArray = [];
+        $returnArray[] = (null === $orderBy || 'ASC' === $orderBy) ? 'ASC' : 'DESC';
+        $returnArray[] = ('ASC' === $returnArray[0]) ? 'down' : 'up';
```
Comment (julienmilazzo) [inline project/src/Twig/AppExtension.php]:
```
return (null === $orderBy || 'ASC' === $orderBy) ? 'down' : 'up';
```

**Review by julienmilazzo — APPROVED** [review]

**Pull request merged**